File: Core/Entities/Match.cs
using Microsoft.EntityFrameworkCore;

namespace Core.Entities
{
    [PrimaryKey(nameof(MatchId))]
    public class Match
    {
        public string MatchId { get; set; }
        public DateTime MatchDate { get; set; }
        public int QueueId { get; set; }
        public string ParticipantPuuids { get; set; }
        public List<MatchParticipant> MatchParticipants { get; set; } = new List<MatchParticipant>();
    }
}

File: client/src/models/entities/Match.ts
import { MatchParticipant } from "./MatchParticipant";

export interface Match {
    matchId: string;
    matchDate: string;
    queueId: number;
    participantPuuids: string;
    region: string;
    matchParticipants: MatchParticipant[];
}